app/src/main/java/com/ece493/group5/adjustableaudio/ui/hearing_test/HearingTestResultFragment.java

```diff
@@ -0,0 +1,423 @@
+
+        audioGramPlot.setDomainStep(StepMode.INCREMENT_BY_VAL, FREQUENCY_INDEX_INCR);
+        audioGramPlot.setDomainBoundaries(FREQUENCY_INDEX_MIN, FREQUENCY_INDEX_MAX, BoundaryMode.FIXED);
+        audioGramPlot.setRangeBoundaries(DBHL_MIN, DBHL_MAX, BoundaryMode.FIXED);
+        audioGramPlot.setRangeStep(StepMode.INCREMENT_BY_FIT, DBHL_INCR);
+
+        audioGramPlot.getGraph().getLineLabelStyle(XYGraphWidget.Edge.TOP).setFormat(new Format()
+        {
+            @Override
+            public StringBuffer format(Object obj, StringBuffer toAppendTo, FieldPosition pos)
+            {
+                int i = Math.round(((Number) obj).floatValue());
+                if (i == 7 | i == 0){
+                    return toAppendTo.append("");
+                }
+                return toAppendTo.append(PLOT_FREQUENCIES[i]);
+            }
+            @Override
+            public Object parseObject(String source, ParsePosition pos)
+            {
+                return null;
+            }
+        });
+
+        audioGramPlot.getGraph().getLineLabelStyle(XYGraphWidget.Edge.LEFT).setFormat(new Format()
+        {
+            @Override
+            public StringBuffer format(Object obj, StringBuffer toAppendTo, FieldPosition pos)
+            {
+                int i = Math.round(((Number) obj).floatValue());
+                return toAppendTo.append(flipPlotRangeValue(i));
+            }
+            @Override
+            public Object parseObject(String source, ParsePosition pos)
+            {
+                return null;
+            }
+        });
+    }
+
+    private int flipPlotRangeValue(int value)
+    {
+        return ((value * -1) + 90);
+    }
+
+    private double flipPlotRangeValue(double value)
+    {
+        return ((value * -1) + 90);
+    }
+
+    private void shareTestResult()
+    {
+        Bitmap audiogramBitmap = createAudioGramBitmap();
+        Uri audiogramUri = createUriFromBitmap(audiogramBitmap);
+        Intent sendIntent =new Intent(android.content.Intent.ACTION_SEND);
+        sendIntent.setType("image/jpeg");
+        sendIntent.putExtra(Intent.EXTRA_STREAM, audiogramUri);
+        startActivity(Intent.createChooser(sendIntent,"Share via"));
+    }
+
+    private Bitmap createAudioGramBitmap()
+    {
+        audioGramPlot.setDrawingCacheEnabled(true);
+        audioGramPlot.measure(audioGramPlot.getWidth(), audioGramPlot.getHeight());
+        Bitmap audiogramBitmap = Bitmap.createBitmap(audioGramPlot.getDrawingCache());
+        audioGramPlot.setDrawingCacheEnabled(false);
+        return audiogramBitmap;
+    }
+
+    private Uri createUriFromBitmap(Bitmap bmp)
+    {
+        try {
+            File audiogramDir = new File(getActivity().getExternalFilesDir(null),
+                    AUDIOGRAM_PATH);
+            if (!audiogramDir.exists())
+            {
+                audiogramDir.mkdir();
+            }
+            File file = new File(audiogramDir, "shared_audiogram_image.jpg");
```

Comment:
Wouldn't this name overwrite other audiogram images? Since it looks like they're all sharing the same directory

Reply:
That's the intention here. We don't store the audiogram images, and instead generate the graph each time the result fragment is opened

That file is used solely for sharing. The way android ACTION_SEND intents work for sending pictures is by putting a file into external storage and sharing the Uri with the chosen app. So each time we share, we just overwrite that path with the picture we want the chosen app to see!